widgetssdk/src/main/java/com/glia/widgets/view/configuration/OptionButtonConfiguration.java

```diff
@@ -0,0 +1,182 @@
+                this.highlightedLayer = prepareDefaultLayerConfiguration();
+            }
+            return new OptionButtonConfiguration(this);
+        }
+
+        private LayerConfiguration prepareDefaultLayerConfiguration() {
+            return new LayerConfiguration.Builder().build();
+        }
+
+        private TextConfiguration prepareDefaultTextConfiguration(ResourceProvider resourceProvider, int textColor) {
+            return new TextConfiguration.Builder()
+                    .textColor(resourceProvider.getColorStateList(textColor))
+                    .textSize(resourceProvider.getDimension(R.dimen.glia_survey_default_text_size))
+                    .build();
+        }
+    }
+
+    /* BEGIN: Parcelable related */
+    @Override
+    public int describeContents() {
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
+        dest.writeParcelable(this.normalText, flags);
+        dest.writeParcelable(this.normalLayer, flags);
+        dest.writeParcelable(this.selectedText, flags);
+        dest.writeParcelable(this.selectedLayer, flags);
+        dest.writeParcelable(this.highlightedText, flags);
+        dest.writeParcelable(this.highlightedLayer, flags);
+    }
+
+    public void readFromParcel(Parcel source) {
+        this.normalText = source.readParcelable(TextConfiguration.class.getClassLoader());
+        this.normalLayer = source.readParcelable(LayerConfiguration.class.getClassLoader());
+        this.selectedText = source.readParcelable(TextConfiguration.class.getClassLoader());
+        this.selectedLayer = source.readParcelable(LayerConfiguration.class.getClassLoader());
+        this.highlightedText = source.readParcelable(TextConfiguration.class.getClassLoader());
+        this.highlightedLayer = source.readParcelable(LayerConfiguration.class.getClassLoader());
+    }
+
+    protected OptionButtonConfiguration(Parcel in) {
+        this.normalText = in.readParcelable(TextConfiguration.class.getClassLoader());
+        this.normalLayer = in.readParcelable(LayerConfiguration.class.getClassLoader());
+        this.selectedText = in.readParcelable(TextConfiguration.class.getClassLoader());
+        this.selectedLayer = in.readParcelable(LayerConfiguration.class.getClassLoader());
+        this.highlightedText = in.readParcelable(TextConfiguration.class.getClassLoader());
+        this.highlightedLayer = in.readParcelable(LayerConfiguration.class.getClassLoader());
+    }
+
+    public static final Creator<OptionButtonConfiguration> CREATOR = new Creator<OptionButtonConfiguration>() {
+        @Override
+        public OptionButtonConfiguration createFromParcel(Parcel source) {
+            return new OptionButtonConfiguration(source);
+        }
+
+        @Override
+        public OptionButtonConfiguration[] newArray(int size) {
+            return new OptionButtonConfiguration[size];
+        }
+    };
+    /* END: Parcelable related */
+}
```

Comment:
new line at end of file

Reply:
@deniss-mc Ok, I'll add it.

But do we really need it in Android? Can you give examples of problems processing the last line of a file if it isn't newline terminated in Android? I'm asking because [as far as I know](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) it's mostly related to Unix tools.

Besides, do you have a list of code style rules agreed for android-sdk-widgets project?

Reply:
To be less specific its just a Github-s complaint (puts marker there if its missing)

List of code style rules will be merged to develop @jakob-mc is dealing with it at the moment (PR is up and discussion available in slack)

Reply:
> To be less specific its just a Github-s complaint (puts marker there if its missing)

Reply:
> To be less specific its just a Github-s complaint (puts marker there if its missing)

Could you please point me to what you are referring to? I can't find such rules in the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html).

> List of code style rules will be merged to develop @jakob-mc is dealing with it at the moment (PR is up and discussion available in slack)

As far as I remember, we discussed only enabling the "Optimize imports" and "Reformat code" options. And these settings doesn't seem to add a new line from what I've tested.

Reply:
you can enable that in the AS settings.

<img width="997" alt="Screenshot 2022-04-27 at 12 14 46" src="https://user-images.githubusercontent.com/49229695/165485136-c3b15157-b5f2-435b-9334-c84455c7a03c.png">

Reply:
@jakob-mc @deniss-mc Yes, I know about that setting in Android Studio. My questions were:
1. Do you have a complete list of code style rules agreed for android-sdk-widgets project so that all team members knew them?
2. Do we really need to add a new line at the end of a file in Android (just out of interest, example of use case or official style guides)?

Reply:
1. I think it's not a part of a specific language code style ruleset but a more general convention. AS formatter doesn't seem to have way to enable that either. Regarding the ruleset we would use - quick google didn't give me clear answer, what's the ruleset AS is using by default. I'll do some research later and document the findings. For now, we would use the rules that AS provides us. 

2. The whole team must choose one or other way to go. Pro to using newline at the end of file is to get rid of the Github complaint.

Reply:
another pro to using the newline is described there -> https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log#comment35216583_5813311

Reply:
I'd say its not even Android related but overall related on how you threat a file. Github uses diff system so each file with any source code is treated as text file. By POSIX definition 
Line:
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206
Incomplete line:
https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_195

=> To make line complete You need to end file with new line

Reply:
Ok, that makes sense, agree.
If you don't find the ruleset AS is using by default, as a last resort, you can check actual rules in AS preferences (for a clean AS installation):

<img width="1118" alt="Screenshot 2022-04-27 at 13 13 53" src="https://user-images.githubusercontent.com/57521863/165496939-dc365c2e-0b8b-4b36-8326-1b9218a879ff.png">